avr-hal-generic/src/i2c.rs

```diff
@@ -497,9 +592,27 @@ macro_rules! impl_twi_i2c {
                 let last = buffer.len() - 1;
                 for (i, byte) in buffer.iter_mut().enumerate() {
```

Comment:
Perhaps I'm blind, but I do not see where the buffer or the byte are being used...isn't the idea that we are reading the bytes out of the TWI and placing into the buffer? I don't see where that is happening. It _looks_ like we are just cycling through the buffer and just signalling that we can consume the next byte...

Reply:
Last line of the loop, the received byte is written into the buffer:

https://github.com/Rahix/avr-hal/blob/9f9d6972e4ac7c149e2644bbc52c47cca81838f4/avr-hal-generic/src/i2c.rs#L415

Reply:
Ah, I _was_ blind. Thx :-)